### PR TITLE
SideMenu.Header: when onClick is not set change the header to be type of div

### PR DESCRIPTION
**Pull request by mozesoron**
### What changed

...

### Why it changed

...

---

- [ ] Change is tested
- [ ] Change is documented
- [ ] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

File: src/SideMenu/README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Container for your header (or anything you want)
 
 | propName | propType | defaultValue | isRequired | description |
 | -        | -        | -            | -          | -           |
-| onClick  | func     | -            | -          | -           |
+| onClick  | func     | -            | -          | header will be of type: button|
```
Comment (lbelinsk) [inline src/SideMenu/README.md]: please elaborate a bit more here. something like:
`A callback that will be called when the header is clicked. when no callback is passed, the Header will be wrapped by a div, otherwise, it will be a clickable button.

File: src/SideMenu/core/Header.js [hunk 2]
```diff
@@ -2,15 +2,21 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import styles from './styles.scss';
 
-const Logo = ({onClick, children}) =>
-  <button type="button" onClick={onClick} className={styles.logo} data-hook="menu-header">
-    {children}
-  </button>;
+const Logo = ({onClick, children}) => {
```
Comment (lbelinsk) [inline src/SideMenu/core/Header.js]: Nice idea to handle your use case.
Please make sure it is approved by our UX. I see you handled the cursor nicely, so I don't see any reason why not to allow this.

File: src/SideMenu/core/Header.js [hunk 3]
```diff
@@ -2,15 +2,21 @@ import React from 'react';
...
+  if (onClick) {
+    return (<button type="button" onClick={onClick} className={styles.logo} data-hook="menu-header">
+      {children}
+    </button>);
+  } else {
+    return (<div className={styles.simpleLogo} data-hook="menu-header">
+      {children}
+    </div>);
+  }
+};
```
Comment (lbelinsk) [inline src/SideMenu/core/Header.js]: Missing a test for this

File: src/Tabs/Tabs.driver.js [hunk 4]
```diff
@@ -17,6 +17,7 @@ const tabsDriverFactory = ({element, wrapper, component}) => {
     getTitles: () => getItems().map(item => item.textContent),
```
Comment (lbelinsk) [inline src/Tabs/Tabs.driver.js]: For next time: This is not related to this pr, better to create a separate one in case we would need to revert

File: src/Tabs/Tabs.driver.js [hunk 5]
```diff
@@ -17,6 +17,7 @@ const tabsDriverFactory = ({element, wrapper, component}) => {
...
     clickTabAt: index => ReactTestUtils.Simulate.click(getItems()[index]),
     getActiveTabIndex: () => getItems().findIndex(item => item.classList.contains(styles.active)),
+    getActiveTabId: () => getItems().filter(item => item.classList.contains(styles.active))[0].id,
```
Comment (lbelinsk) [inline src/Tabs/Tabs.driver.js]: maybe it should be safer. if there is no active tab, this will throw: can't access id of undefined or something...

File: src/Tabs/core/TabItem/index.js [hunk 6]
```diff
@@ -24,6 +24,7 @@ class TabItem extends React.Component {
 
     const containerProps = {
       key: item.id,
+      id: item.id,
```
Comment (lbelinsk) [inline src/Tabs/core/TabItem/index.js]: Need to update the proptypes here: https://github.com/mozesoron/wix-style-react/blob/e1a167aa7b3b2da09eb67aebf390bd1dc5413874/src/Tabs/core/constants/tab-prop-types.js

File: src/Tabs/Tabs.driver.js (hunk 4 above)
Comment (mozesoron) [reply src/Tabs/Tabs.driver.js]: you are right I thought that it will let me create another pr.

File: src/Tabs/core/TabItem/index.js (hunk 6 above)
Comment (mozesoron) [reply src/Tabs/core/TabItem/index.js]: I already see an id is set under the item const: https://github.com/mozesoron/wix-style-react/blob/e1a167aa7b3b2da09eb67aebf390bd1dc5413874/src/Tabs/core/constants/tab-prop-types.js#L8
 do I need to change anything else?

File: src/SideMenu/core/Header.js (hunk 3 above)
Comment (mozesoron) [reply src/SideMenu/core/Header.js]: added 2 tests

File: src/Tabs/core/TabItem/index.js (hunk 6 above)
Comment (lbelinsk) [reply src/Tabs/core/TabItem/index.js]: No, I missed that, all good then

**Comment by lbelinsk** [comment]
There is an issue that eyes helped us catching here.
If a user didn't give onClick then we must make sure that the dom layout would still look the same, only the cursor should change.

Eyes shows us that there is a diff with the margins / paddings

image

**Comment by lbelinsk** [comment]
@mozesoron  I am closing this in favor of https://github.com/wix/wix-style-react/pull/1900
Please feel free to say if you think this is not the way to go, and we could then reopen this
